os/arch/arm/src/amebasmart/amebasmart_pmhelpers.c

```diff
@@ -115,19 +115,29 @@ int SOCPS_AONWakeReason(void)
 
 void pg_timer_int_handler(void *Data)
 {
-	pmvdbg("Timer wakeup interrupt handler!!\n");
+	pmvdbg("PM Timer interrupt handler!!\n");
+	switch (g_pm_timer.timer_type) {
+		case PM_LOCK_TIMER:
+			pm_relax(PM_IDLE_DOMAIN, PM_NORMAL);
+			break;
+		case PM_WAKEUP_TIMER:
+			pm_activity(PM_IDLE_DOMAIN, 9);
+			break;
+		default:
+			pmdbg("Timer callback triggered without setting timer, Unexpected Error!!!\n");
+			break;
+	}
 	// Reset the global struct
-    g_timer_wakeup.use_timer = 0;
-    g_timer_wakeup.timer_interval = 0;
+	g_pm_timer.timer_type = 0;
+	g_pm_timer.timer_interval = 0;
 	// Switch status back to normal mode after wake up from interrupt
-	pm_activity(PM_IDLE_DOMAIN, 9);
 }
 
-void ap_timer_helper(void) {
+void up_set_timer(void) {
 	// Check whether timer interrupt need to be set
-	if (g_timer_wakeup.use_timer) {
+	if (g_pm_timer.timer_type) {
 		gtimer_init(&g_timer1, TIMER1);
-		gtimer_start_one_shout(&g_timer1, g_timer_wakeup.timer_interval, (void *)pg_timer_int_handler, NULL);
+		gtimer_start_one_shout(&g_timer1, g_pm_timer.timer_interval, (void *)pg_timer_int_handler, NULL);
```

Comment:
In AIDual, timer1~13 is general purpose timer free for used, and it depends on your choice, I just simply picked TIMER1. In other words, there is no relation between functionality usage and timer index. It only depends on your use case and scenarios.
If needed, you can change the timer index according to your preferences.

Reply:
Additionally, we have a timer porting layer https://github.com/Samsung/TizenRT/blob/master/os/arch/arm/src/amebasmart/amebasmart_timer_lowerhalf.c, but I'm not sure whether Samsung side is utilizing this using FD from application layer for timer control? 
If yes, do we still invoke timer like this way (ie. invoking BSP code directly)?

Reply:
@amogh-samsung Currently, we can know which timer is used when we search entire codes. I mean user can't know TIMER1 is available or not. Actually it's unavailable because pm uses. 
In short-term, leave a comment on timer or somewhere (pm?) that pm uses TIMER1 on RTL8730E.
In long-term, we should make a timer resource manager to allocate free timer. (This is not related to PM)

Reply:
Understood,
@edwakuwaku does the use of the porting layer help with this?

Reply:
It should be similar as to UART registration, and we can set how many of them need to be utilised, using defconfig and macros. Since we have 13 timers for your use case, we can just assigned each as to specific use case as for now, which is the "short term" mentioned by @sunghan-chang . If in the future, we cannot cover all the use case, let's go for timer resource manager. 
I have one question, does the TP1x use this timer porting layer before?